py-polars/polars/lazyframe/frame.py

```diff
@@ -1994,6 +1995,88 @@ def cache(self) -> Self:
         """Cache the result once the execution of the physical plan hits this node."""
         return self._from_pyldf(self._ldf.cache())
 
+    def cast(
+        self,
+        dtypes: Mapping[ColumnNameOrSelector, PolarsDataType] | PolarsDataType,
+        *,
+        strict: bool = True,
+    ) -> Self:
+        """
+        Cast LazyFrame column(s) to the specified dtype(s).
+
+        Parameters
+        ----------
+        dtypes
+            Mapping of column names (or selector) to dtypes, or a single dtype
+            to which all columns will be cast.
+        strict
+            Throw an error if a cast could not be done (for instance, due to an
+            overflow).
+
+        Examples
+        --------
+        >>> from datetime import date
+        >>> lf = pl.LazyFrame(
+        ...     {
+        ...         "foo": [1, 2, 3],
+        ...         "bar": [6.0, 7.0, 8.0],
+        ...         "ham": [date(2020, 1, 2), date(2021, 3, 4), date(2022, 5, 6)],
+        ...     }
+        ... )
+
+        Cast specific frame columns to the specified dtypes:
+
+        >>> lf.cast({"foo": pl.Float32, "bar": pl.UInt8}).collect()
+        shape: (3, 3)
+        ┌─────┬─────┬────────────┐
+        │ foo ┆ bar ┆ ham        │
+        │ --- ┆ --- ┆ ---        │
+        │ f32 ┆ u8  ┆ date       │
+        ╞═════╪═════╪════════════╡
+        │ 1.0 ┆ 6   ┆ 2020-01-02 │
+        │ 2.0 ┆ 7   ┆ 2021-03-04 │
+        │ 3.0 ┆ 8   ┆ 2022-05-06 │
+        └─────┴─────┴────────────┘
+
+        Cast all frame columns to the specified dtype:
+
+        >>> lf.cast(pl.Utf8).collect().to_dict(False)
+        {'foo': ['1', '2', '3'],
+         'bar': ['6.0', '7.0', '8.0'],
+         'ham': ['2020-01-02', '2021-03-04', '2022-05-06']}
+
+        Use selectors to define the columns being cast:
+
+        >>> import polars.selectors as cs
+        >>> lf.cast({cs.numeric(): pl.UInt32, cs.temporal(): pl.Utf8}).collect()
+        shape: (3, 3)
+        ┌─────┬─────┬────────────┐
+        │ foo ┆ bar ┆ ham        │
+        │ --- ┆ --- ┆ ---        │
+        │ u32 ┆ u32 ┆ str        │
+        ╞═════╪═════╪════════════╡
+        │ 1   ┆ 6   ┆ 2020-01-02 │
+        │ 2   ┆ 7   ┆ 2021-03-04 │
+        │ 3   ┆ 8   ┆ 2022-05-06 │
+        └─────┴─────┴────────────┘
+
+        """
+        if not isinstance(dtypes, Mapping):
+            return self.select(F.all().cast(dtypes, strict=strict))
+        else:
+            from polars.selectors import expand_selector
+
+            cast_map = {}
+            for c, dtype in dtypes.items():
+                dtype = py_type_to_dtype(dtype)
+                cast_map.update(
+                    {c: dtype}
+                    if isinstance(c, str)
+                    else {x: dtype for x in expand_selector(self, c)}
+                )
```

Comment:
My brain always has difficulty parsing this ternary syntax... isn't this more readable?

```suggestion
                if isinstance(c, str):
                    cast_map[c] = dtype
                else:
                    cast_map.update({x: dtype for x in expand_selector(self, c)})
```

Reply:
Not to me; I actually find the ternary construct cleaner there, but I guess that's down to personal taste and/or which languages you're most attuned to - I can change it though 😅

Reply:
In that case, let's leave it as-is!